encode_test.go

```diff
@@ -104,6 +106,78 @@ func TestEncodePartQuotedPrintableHeaders(t *testing.T) {
 	test.DiffGolden(t, b.Bytes(), "testdata", "encode", "part-quoted-printable-headers.golden")
 }
 
+type oneByOneReader struct {
```

Comment:
Please add comment on the purpose of this to make tests easier to understand

Reply:
done